backend-scripts/download_and_update_routes_details.py

```diff
@@ -0,0 +1,86 @@
+#!venv/bin/python
+# -*- coding: utf-8 -*-
+
+import sqlite3
+import requests
+import re
+import html
+
+conn = sqlite3.connect('../generated/routes.db')
+conn.row_factory = sqlite3.Row
+
+headers = {'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'X-Requested-With': 'XMLHttpRequest'}
+
+# Download urls are always one of four formats, Only route type matters.
+# https://gyms.vertical-life.info/en/gaswerk-greifensee/iframe/64264/iframe_route_detail
+# https://gyms.vertical-life.info/en/gaswerk-greifensee/iframe/126018/iframe_boulder_detail
+# https://gyms.vertical-life.info/en/gaswerk-schlieren/iframe/64264/iframe_route_detail
+# https://gyms.vertical-life.info/en/gaswerk-schlieren/iframe/126018/iframe_boulder_detail
+route_details_url = "https://gyms.vertical-life.info/en/gaswerk-%s/iframe/%s/iframe_%s_detail"
+
+# Match name whic is inside <h4> tag.
+# <h4>Feedbackkulturparadigmenwechsel 6a+<\/h4>
+# Additional patterns are needed to filter out
+# - route/boulder grade which is included on the details page in the title
+# - label for not zlagged routes
+# <h4>Download the Vertical-Life Climbing app and mark your climbs!</h4>
+name_pattern = re.compile('.*route-info.*<h4>(.+?) \S*<\\\\/h4>.*route-topo.*')
+
+# Match first jpg image on the details page which is after 'holder' div and before 'wall-holder'
+# <div id="holder" ... data-image-src="https://d1ffqbcmevre4q.cloudfront.net/e90e4c8a080771d0028b7f753e46e45e.jpg">...<div id="wall-holder"...
+sector_url_pattern = re.compile('.*<div id=\\\\"holder\\\\".*data-image-src=\\\\"https://d1ffqbcmevre4q\.cloudfront\.net/(.+?\.jpg)\\\\">.*id=\\\\"wall-holder\\\\".*')
```

Comment:
Impressive regex knowledge. :)

Reply:
You mean: impressive google skills. :D
You see only the last iteration, the first run ended with all routes having the name of the map image saved.